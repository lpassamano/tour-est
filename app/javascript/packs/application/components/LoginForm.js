import React, { Component } from "react";
import { navigate } from "@reach/router";
import PropTypes from "prop-types";

class LoginForm extends Component {
  static defaultProps = { navigate };
  state = { username: "", password: "" };

  handleSubmit = async event => {
    event.preventDefault();
    const result = await this.props.onLogin(
      this.state.username,
      this.state.password
    );

    if (result.ok) {
      this.props.navigate("/admin");
    } else {
      console.error(result.data.error);
    }
  };

  handleChange = event => {
    const { name, value } = event.target;
    this.setState({ [name]: value });
  };

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <input
          name="username"
          type="text"
          id="username"
          value={this.state.username}
          onChange={this.handleChange}
        />
        <input
          name="password"
          type="password"
          id="password"
          value={this.state.password}
          onChange={this.handleChange}
        />
        <button type="submit">Log In</button>
      </form>
    );
  }
}

LoginForm.propTypes = {
  navigate: PropTypes.func.isRequired,
  onLogin: PropTypes.func.isRequired
};

export default LoginForm;
