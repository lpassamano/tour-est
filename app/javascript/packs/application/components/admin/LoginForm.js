import React, { Component } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";

import * as staffUserActions from "../../redux/staffUser/actions";

export class LoginForm extends Component {
  state = { username: "", password: "" };

  handleSubmit = event => {
    event.preventDefault();
    this.props.onLogin(this.state.username, this.state.password);
  };

  handleChange = event => {
    const { name, value } = event.target;
    this.setState({ [name]: value });
  };

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <label htmlFor="username">Username: </label>
        <input
          name="username"
          type="text"
          value={this.state.username}
          onChange={this.handleChange}
        />
        <br />
        <label htmlFor="password">Password: </label>
        <input
          name="password"
          type="password"
          value={this.state.password}
          onChange={this.handleChange}
        />
        <br />
        <button type="submit">Log In</button>
      </form>
    );
  }
}

LoginForm.propTypes = {
  onLogin: PropTypes.func.isRequired
};

const mapDispatchToProps = {
  onLogin: staffUserActions.loginStaffUser
};

const enhance = connect(
  null,
  mapDispatchToProps
);

export default enhance(LoginForm);
