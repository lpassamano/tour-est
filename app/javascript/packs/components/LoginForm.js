import React, { Component } from 'react'
import { create } from 'apisauce'

const api = create({
  baseURL: '/',
  headers: { 'x-csrf-token': document.querySelector("[name=csrf-token]").getAttribute("content") }
})

class LoginForm extends Component {
  state = { username: "", password: "" }

  handleSubmit = async (event) => {
    event.preventDefault();
    // const token = document.querySelector("[name=csrf-token]").getAttribute("content");
    // console.log(token);
    // this.setState({ [csrfToken]: token })
    // console.log(this.state)
    const result = await api.post("/sessions", this.state)
    if (result.ok) {
      console.log("Login successful!!!!", result.data)
    } else {
      console.error(result)
    }
  }
  // document.querySelector("[name=csrf-token]").getAttribute("content")

  handleChange = (event) => {
    const { name, value } = event.target
    this.setState({ [name]: value })
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <input name="username" type="text" value={this.state.username} onChange={this.handleChange} />
        <input name="password" type="password" value={this.state.password} onChange={this.handleChange} />
        <button type="submit">Log In</button>
      </form>
    );
  }
}

export default LoginForm;
