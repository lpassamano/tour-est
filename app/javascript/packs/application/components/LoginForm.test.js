import LoginForm from "./LoginForm";
import { shallow, mount } from "enzyme";
import React from "react";

describe("<LoginForm />", () => {
  const setup = ({ onLogin = jest.fn(), navigate = jest.fn() }) => {
    const component = mount(
      <LoginForm onLogin={onLogin} navigate={navigate} />
    );
    const event1 = { target: { name: "username", value: "leigh" } };
    const event2 = { target: { name: "password", value: "123abc" } };
    component.find("input#username").simulate("change", event1);
    component.find("input#password").simulate("change", event2);
    return component;
  };

  test("onSubmit - when submit is clicked the current values of username and password are submitted", done => {
    const onLogin = jest.fn();
    const navigate = to => {
      expect(to).toEqual("/admin");
      done();
    };
    onLogin.mockResolvedValue({ ok: true });
    const component = setup({ onLogin, navigate });
    component.find("form").simulate("submit", { preventDefault: () => null });
    expect(onLogin).toHaveBeenCalledWith("leigh", "123abc");
  });

  test("handleChange - when text is typed into the email and password fields the state is updated", () => {
    const component = setup({});
    expect(component.state("username")).toEqual("leigh");
    expect(component.state("password")).toEqual("123abc");
  });

  test("render", () => {
    const component = setup({});
    expect(component).toMatchSnapshot();
  });
});
