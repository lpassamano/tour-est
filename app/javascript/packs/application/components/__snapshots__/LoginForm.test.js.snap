// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> render 1`] = `
<LoginForm
  navigate={[Function]}
>
  <form
    onSubmit={[Function]}
  >
    <input
      id="username"
      name="username"
      onChange={[Function]}
      type="text"
      value="leigh"
    />
    <input
      id="password"
      name="password"
      onChange={[Function]}
      type="password"
      value="123abc"
    />
    <button
      type="submit"
    >
      Log In
    </button>
  </form>
</LoginForm>
`;
